modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageManagerImpl.java

```diff
@@ -872,6 +884,24 @@ public CompletableFuture<IndexWithTerm> raftNodeIndex() {
         }));
     }
 
+    @Override
+    public CompletableFuture<Void> becomeLonelyLeader(boolean pauseLeaderSecondaryDuties) {
```

Comment:
Can this method be called in parallel?
If not, then everything is fine. If yes, then we need to protect it somehow.

Reply:
Right now it can't due to the guarantees of the networking. But it's easy to break them (as they are subtle and implicit), so I added an explicit synchronization. This should not be a problem as the method does not block for a long time and it is getting called really rarely.

Reply:
We like to turn asynchronous code into synchronous =)